okhttp/src/main/java/okhttp3/internal/ws/WebSocketWriter.java

```diff
@@ -173,6 +179,9 @@ void writeMessageFrame(int formatOpcode, long byteCount, boolean isFirstFrame,
     if (isFinal) {
       b0 |= B0_FLAG_FIN;
     }
+    if (compressionEnabled && isFirstFrame) {
```

Comment:
I think we just want a parameter like `boolean compressionEnabled`.

Reply:
Yep. I went for `isCompressed`, looks like it better describes the fact that it's about this exact message, as opposed to some global state.